README.md
# Python code

Thank you for your interest in this project repository that we hope will be useful. This README document is a summary of how we (the coders) are going to analyse the Heart Disease Dataset, which was produced as a result of a study conducted in the late 1980s by the Hungarian Institute of Cardiology (Budapest), the Zurich and Basel University Hospitals (Switzerland), and the Long Beach and Cleveland Clinic Foundation (Long Beach, United States of America). The following points describe the main characteristics of the analysis plan:

# Diagnostic Models for Heart Disease Prediction

The aim of project is to predict whether a person with some particular characteristics, such as their cholesterol and the type of angina they experience, will develop a cardiovascular disease which could possibly be used to design interventions to prevent/lower the risk of cardiovascular disease.

# Installation

Install the Python interpreter correctly on your system.

# Data and file overview
The data is obtained from four databases from Cleveland, Hungary, Switzerland, and the VA Long Beach.

This database contains 76 attributes, but all published experiments refer to using a subset of 14 of them which will be used for our analysis as well.

## Variables Characteristics
1. Age: The person’s age in years
2. Sex: The person’s gender (0 = female, 1 = male)
3. Cp: Chest pain type: 
   Value 1: Typical angina;
   Value 2: Atypical angina;
   Value 3: Non-angina pain;
   Value 4: Asymptomatic;
4. Trestbps: The person’s resting blood pressure (in mm Hg on admission to the hospital)
5. Chol: The person’s cholesterol measured in mg/dl
6. Fbs: The person’s fasting blood sugar (fasting blood sugar > 120 mg/dl; 0 = False, 1 = True)
7. Restecg: The person’s resting electrocardiographic results:
   Value 0: Normal;
   Value 1: Having ST-T wave abnormality (T wave inversions and/or ST elevation or depression of > 0.05 mV);
   Value 2: Showing probable or definite left ventricular hypertrophy by Estes' criteria
8. Thalach: The maximum heartrate the person can achieve
9. Exang: Exercise induced angina (0 = No, 1 = Yes)
10. Oldpeak: ST depression induced by exercise relative to rest
11. Slope: The slope of the peak exercise ST segment:
 Value 1: Upsloping;
 Value 2: Flat;
 Value 3: Downsloping
12. Ca: The number of major blood vessels (0-3)
13. Thal: Sortcut for a blood disorder called Thalassemia:
 Value 3: Normal;
 Value 6: Fixed defect;
 Value 7: Reversable defect
14. Num: The diagnosis of heart disease:
 Value 0: < 50% diameter narrowing (No heart disease);
 Value 1: > 50% diameter narrowing (Heart disease)

# Analysis and utput
1. Descriptive Statistics of variables considered important for the study

   - Split  between quantitative and qualitative, bar charts, correlation heatmap describing and summarizing various data points and their relation to other data points so that any patterns that exit may become clear.

2. Feature Selection

   - This address the relationship between different factors and heart disease

3. Machine Learning

   - Logistic Regression (to access the probability of the given variable, which in this case was whether the person suffers from some type of cardiovascular disease based on the 13 factors that were studied) 

   - Naïve Bayes (to make quick predictions regarding possibility of suffering cardiovascular disease)

   - K-means (to group the given data and discover underlying patterns)

# Note:
This is an ongoing effort so more updates and details will be addeda as the project progresses. Please check the file ROADMAP.md for more details on milestones and planned tasks.

# License

[MIT](https://choosealicense.com/licenses/mit/)


CHANGES.md
### Week 1

Introduction and Methods were added



### Week 2

- [x] An implementation of code for Descriptive Statistics analysis has been uploaded.

- [x] The structure of the code was improved using the principles of abstraction and decomposition.
    - As our main goal was plotting descriptive statistics we identified that our data had only two kinds of variables which needed to be ploted differently. Therefore, we separated the variables in two categories. The key point was plot different kinds of plot for different kinds of variables and names/data from individual variable was added detail. So, we wrote a function that plots different kind of plots for different knid of variables. Within the function we decomposed the code so it would focus on individual variables belonging to different categories using a for loop. This shortened our code and we could avoid writing the same code for plotting data multiple time. Further it ensured that all the plots of the same kind looked the same
    





test.py
"This is a test".


Descriptive Statistics.py
# -*- coding: utf-8 -*-
"""
Created on Thu May 11 12:47:07 2023

@author: Aparna Pandey, Spyridoula Georgiou
"""
# Load required modules
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns

# Load the data as dataframe using pandas
df = pd.read_csv("ProcessedClevelandData.csv")

# Print a concise summary of a DataFrame.
# This method prints information about a DataFrame including the index dtype
# and columns, non-null values and memory usage.

df.info()

# Check for missing data in the dataframe for quantitative
Missing_data_count = df.isna().sum()
print("No. of missing data points in different columns:\n",Missing_data_count)

# In case of missing data impute the data based on the most common value in the
# column
for i in df.columns:
    column_index = df.columns.get_loc(i)
    mode = df.mode().iloc[:,column_index]
    print(mode[0])
    df[i] = df[i].fillna(mode[0])

# Assess the summary of aggregate measures for the data
df.describe()

# We need to see if there any outliers in the dataset for continuous variables
# we check that based on the quartiles.

continous_features = ['age','trestbps','chol','thalach','oldpeak']

#Calculate and identify outliers
def detect_outliers(df_out):
    """Returned the number of outliers for continuous variable based on 
    interquartile range.
    """
    for each_feature in df_out.columns:
        feature_data = df_out[each_feature]
        q_1 = np.percentile(feature_data, 25.)
        q_3 = np.percentile(feature_data, 75.)
        iqr = q_3 - q_1
        outlier_step = iqr
        outliers = feature_data[~((feature_data >= q_1 - outlier_step) \
                & (feature_data <= q_3 + outlier_step))].index.tolist()
        print(f"For the feature {each_feature}, the number of outliers is {len(outliers)}.")
detect_outliers(df[continous_features])
###-------------------------------------------------------------------------###
###                     Descriptive Data Analyis                            ###
###-------------------------------------------------------------------------###
#Set the parameters that control the general style of the plots.
sns.set_style('white')
sns.set_palette('twilight')
p1 = sns.countplot(data= df, x='sex',hue='num')
plt.legend(title='"Diagnosis', loc='upper left', labels=['No Heart Disease', 'Heart Disease'])
plt.ylabel('Number of Cases',fontsize=13)
plt.xlabel('Gender',fontsize=13)
p1.set_xticklabels(('Female', 'Male'))
sns.despine()
###-------------------------------------------------------------------------###

p2 = sns.countplot(data= df, x='cp',hue='num')
plt.legend(title='"Diagnosis', loc='upper left', labels=['No Heart Disease','Heart Disease'])
plt.ylabel('Number of Cases',fontsize=13)
plt.xlabel('Chest Pain',fontsize=13)
p2.set_xticklabels(('Typical Angina', 'Atypical Angina','Non - Angina Pain','Asymptomatic'))
sns.despine()
###-------------------------------------------------------------------------###

p3 = sns.countplot(data= df, x='fbs',hue='num',order=[1,0])
plt.legend(title='"Diagnosis', loc='upper left', labels=['No Heart Disease','Heart Disease'])
plt.ylabel('Number of Cases',fontsize=13)
plt.xlabel('Fasting Blood Sugar (>120 mg/dl)',fontsize=13)
p3.set_xticklabels(('True', 'False'))
sns.despine()
###-------------------------------------------------------------------------###

p4 = sns.countplot(data= df, x='restecg',hue='num',order=[1,0,2])
plt.legend(title='"Diagnosis', loc='upper left', labels=['No Heart Disease','Heart Disease'])
plt.ylabel('Number of Cases',fontsize=13)
plt.xlabel('Resting Electrocardiographic Results',fontsize=13)
p4.set_xticklabels(('Normal', 'Abnormal','Ventricular Hypertrophy'))
sns.despine()
###-------------------------------------------------------------------------###

p5 = sns.countplot(data= df, x='exang',hue='num',order=[1,0])
plt.legend(title='"Diagnosis', loc='upper left', labels=['No Heart Disease','Heart Disease'])
plt.ylabel('Number of Cases',fontsize=13)
plt.xlabel('Exercise Induced Angina',fontsize=13)
p5.set_xticklabels(('Yes', 'No'))
sns.despine()
###-------------------------------------------------------------------------###

p6 = sns.countplot(data= df, x='slope',hue='num',order=[3,2,1])
plt.legend(title='"Diagnosis', loc='upper left', labels=['No Heart Disease','Heart Disease'])
plt.ylabel('Number of Cases',fontsize=13)
plt.xlabel('The Slope Of The Peak Exercise',fontsize=13)
p6.set_xticklabels(('Downsloping', 'Flat','Upsloaping'))
sns.despine()
###-------------------------------------------------------------------------###

p7 = sns.countplot(data= df, x='thal',hue='num',order=[6,3,7])
plt.legend(title='"Diagnosis', loc='upper left', labels=['No Heart Disease','Heart Disease'])
plt.ylabel('Number of Cases',fontsize=13)
plt.xlabel('Type of Thalassemia',fontsize=13)
p7.set_xticklabels(('Downsloping', 'Flat','Upsloaping'))
sns.despine()
###-------------------------------------------------------------------------###

p8 = sns.countplot(data= df, x='ca',hue='num',order=[3,2,1,0])
plt.legend(title='"Diagnosis', loc='upper left', labels=['No Heart Disease','Heart Disease'])
plt.ylabel('Number of Cases',fontsize=13)
plt.xlabel('The Number Of Major Blood Vessels',fontsize=13)
p8.set_xticklabels(('3', '2','1','0'))
sns.despine()
###-------------------------------------------------------------------------###

sns.histplot(data=df, x="age", kde=True, bins=20)
plt.title('Distribution of Age')
plt.xlabel('Age')
plt.ylabel('Number of Cases')
sns.despine()
###-------------------------------------------------------------------------###

sns.histplot(data=df, x="chol", kde=True, bins=20)
plt.title('Distribution of Cholesterol')
plt.xlabel('Cholesterol')
plt.ylabel('Number of Cases')
sns.despine()
###-------------------------------------------------------------------------###

sns.histplot(data=df, x="trestbps", kde=True, bins=20)
plt.title('Distribution of Resting Blood Pressure')
plt.xlabel('Resting Blood Pressure')
plt.ylabel('Number of Cases')
sns.despine()
###-------------------------------------------------------------------------###

sns.histplot(data=df, x="thalach", kde=True, bins=20)
plt.title('Distribution of the Maximum Heart Rate')
plt.xlabel('Maximum Heart Rate')
plt.ylabel('Number of Cases')
sns.despine()
###-------------------------------------------------------------------------###

sns.histplot(data=df, x="oldpeak", kde=True, bins=20)
plt.title('Distribution of the ST Depression Induced by Exercise')
plt.xlabel('ST Depression Induced by Exercise')
plt.ylabel('Number of Cases')
sns.despine()
###-------------------------------------------------------------------------###

p=sns.boxplot(data=df,x='num',y='age')
plt.xlabel('Diagnosis')
plt.ylabel('Age')
plt.xlabel('Diag')
p.set_xticklabels(['No Heart Disease', 'Heart Disease'])
###-------------------------------------------------------------------------###

p=sns.boxplot(data=df,x='num',y='trestbps')
plt.xlabel('Diagnosis')
plt.ylabel('Resting Blood Pressure')
p.set_xticklabels(['No Heart Disease', 'Heart Disease'])
###-------------------------------------------------------------------------###

p=sns.boxplot(data=df,x='num',y='chol')
plt.xlabel('Diagnosis')
plt.ylabel('Cholesterol')
p.set_xticklabels(['No Heart Disease', 'Heart Disease'])
###-------------------------------------------------------------------------###

p=sns.boxplot(data=df,x='num',y='thalach')
plt.xlabel('Diagnosis')
plt.ylabel('Maximum Heart Rate Achieved')
p.set_xticklabels(['No Heart Disease', 'Heart Disease'])
###-------------------------------------------------------------------------###

p=sns.boxplot(data=df,x='num',y='oldpeak')
plt.xlabel('Diagnosis')
plt.ylabel('ST Depression Induced By\n Exercise Relative To Rest')
p.set_xticklabels(['No Heart Disease', 'Heart Disease'])
###-------------------------------------------------------------------------###
# Assess the correlation between variables in the dataset. Correlation between
# variables might affect the regression analysis. So, this information will be
# helpful to explain any unexpected results or for model selection
corrmat = df.corr()
matrix = np.triu(corrmat)
fig, ax = plt.subplots(figsize=(12, 12))
sns.heatmap(corrmat, square=True, annot=True, cmap="Purples", mask = matrix)
plt.show()
#save the plot
fig.savefig('./Correlation Heatmap', bbox_inches='tight')
###-------------------------------------------------------------------------###


descriptive_statistics_abstraction_and_decomposition.py
# -*- coding: utf-8 -*-
"""
Created on Thu May 11 12:47:07 2023

@author: Aparna Pandey, Spyridoula Georgiou
"""
# Load required modules
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns

# Load the data as dataframe using pandas
df = pd.read_csv("ProcessedClevelandData.csv")

# Print a concise summary of a DataFrame.
# This method prints information about a DataFrame including the index dtype
# and columns, non-null values and memory usage.

df.info()

# Check for missing data in the dataframe for quantitative
Missing_data_count = df.isna().sum()
print("No. of missing data points in different columns:\n",Missing_data_count)

# In case of missing data impute the data based on the most common value in the
# column
for i in df.columns:
    column_index = df.columns.get_loc(i)
    mode = df.mode().iloc[:,column_index]
    df[i] = df[i].fillna(mode[0])

# Assess the summary of aggregate measures for the data
df.describe()

# We need to see if there any outliers in the dataset for continuous variables
# we check that based on the quartiles.

continous_features = ['age','trestbps','chol','thalach','oldpeak']
categorical_features = ['sex','cp','fbs','restecg','exang', 'slope', 'ca', 'thal']

#Calculate and identify outliers
def detect_outliers(df_out):
    """Returned the number of outliers for continuous variable based on 
    interquartile range.
    """
    for each_feature in df_out.columns:
        feature_data = df_out[each_feature]
        q_1 = np.percentile(feature_data, 25.)
        q_3 = np.percentile(feature_data, 75.)
        iqr = q_3 - q_1
        outlier_step = iqr
        outliers = feature_data[~((feature_data >= q_1 - outlier_step) \
                & (feature_data <= q_3 + outlier_step))].index.tolist()
        print(f"For the feature {each_feature}, the number of outliers is {len(outliers)}.")
detect_outliers(df[continous_features])


###-------------------------------------------------------------------------###
###          Descriptive Data Analyis (abstraction and decomposition)      ###
###-------------------------------------------------------------------------###
#Set the parameters that control the general style of the plots.
sns.set_style('white')
sns.set_palette('twilight')

# Assess the correlation between variables in the dataset. Correlation between
# variables might affect the regression analysis. So, this information will be
# helpful to explain any unexpected results or for model selection
corrmat = df.corr()
matrix = np.triu(corrmat)
fig, ax = plt.subplots(figsize=(12, 12))
sns.heatmap(corrmat, square=True, annot=True, cmap="Purples", mask = matrix)
plt.show()
#save the plot
fig.savefig('./Correlation Heatmap', bbox_inches='tight')
###-------------------------------------------------------------------------###
# Classify the variables based on their characteristic such as categorical, continuous
# numerical, etc. as different plots are used to visualize different kinds of data
# Also indicate which is the traget/ variable of interest or response variable

continous_features = ['age','trestbps','chol','thalach','oldpeak']
categorical_features = ['sex','cp','fbs','restecg','exang', 'slope', 'ca', 'thal']
dependent_variable = df.iloc[:,-1] #target column

# Make a  copy of the dataframe
df2 = df.copy()

# Change the labels in the copy based on conditions and column descriptions(This
# is optional, just to make the plot labels more intutive). We have done this
# in multiple ways just to show that there are different ways to acheive the same
# result.
# Note: We can use the following code to infer the number and names of the categories
# of a categorical variable if it is not known
# List_Of_Categories_In_Column=list(df['name_of_column'].value_counts().index)
# print(List_Of_Categories_In_Column)
df2['sex'] = np.where(df['sex'] == 0, "Female", "Male")

df2['cp'] = df2['cp'].map({1:'ypical Angina',2:'Atypical Angina',\
                           3:'Non - Angina Pain',4:'Asymptomatic'})

def change_fbs(fasting_blood_sugar):
    """Change the fbs categories to descriptive string literals for more 
    intutive understanding of the plots
    """
    if fasting_blood_sugar == 0:
        return 'False'
    return 'True'
df2['fbs'] = df2['fbs'].apply(change_fbs)

df2['restecg'] = df2['restecg'].map({0:'Normal',1:'Abnormal',2:'Ventricular Hypertrophy'})

df2['exang'] = np.where(df['exang'] == 0, "No", "Yes")

df2['slope'] = df2['slope'].map({1:'Upsloaping',2:'Flat',3:'Downsloping'})

df2['thal'] = df2['thal'].map({3:'Normal',6:'Fixed',7:'Reversed'})

# Write a function to plot different kinds of plots for different knids of
# variables

def plot_descriptive_statistics(df_input,continuous_variables,\
                                categorical_variables,response_variable):
    """A general function for plotting descriptive statistics
    """
    #Set the parameters that control the general style of the plots.
    sns.set_style('white')
    sns.set_palette('twilight')
    for each_feature in df_input.columns:
        index_column = df_input.columns.get_loc(each_feature)
        if each_feature in continuous_variables:
            print(each_feature, "is a continuous variable.")
            # code for plotting histograms
            plt.figure(index_column)
            plot_1=sns.histplot(data=df_input, x=each_feature, kde=True, bins=20)
            plt.title(f"Distribution of {each_feature}")
            plt.xlabel(each_feature)
            plt.ylabel('Number of Cases')
            sns.despine()
            figure = plot_1.get_figure()
            figure.savefig(f"Histogram_{each_feature}.png")
            # code for plotting boxplots
            plt.figure(index_column+len(df_input.columns))
            plot_2=sns.boxplot(data=df_input,x=response_variable,y=each_feature)
            plt.xlabel('Diagnosis')
            plt.ylabel(f"{each_feature}")
            plot_2.set_xticklabels(['No Heart Disease', 'Heart Disease'])
            figure = plot_2.get_figure()
            figure.savefig(f"Box_plot_{each_feature}.png")
        elif each_feature in categorical_variables:
            print(each_feature,"is a continuous variable.")
            # Code for plotting count plots
            plt.figure(index_column)
            plot_1= sns.countplot(data= df_input, x=each_feature,hue=response_variable)
            plt.legend(title='"Diagnosis', labels=['No Heart Disease','Heart Disease'],\
                       bbox_to_anchor=(1.02, 1), loc='upper left', borderaxespad=0)
            plt.ylabel('Number of Cases',fontsize=13)
            plt.xlabel(f"{each_feature}",fontsize=13)
            sns.despine()
            figure = plot_1.get_figure()
            figure.savefig(f"Count_plot_{each_feature}.png")
            #plot_1.set_xticklabels(('Downsloping', 'Flat','Upsloaping'))
# Employ the function to generate and save multiple plots at the same time.
plot_descriptive_statistics(df2,continous_features,categorical_features,dependent_variable)


descriptive_statistics.py
# -*- coding: utf-8 -*-
"""
Created on Thu May 11 12:47:07 2023

@author: Aparna Pandey, Spyridoula Georgiou
"""
# Load required modules
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns

# Load the data as dataframe using pandas
df = pd.read_csv("ProcessedClevelandData.csv")

# Print a concise summary of a DataFrame.
# This method prints information about a DataFrame including the index dtype
# and columns, non-null values and memory usage.

df.info()

# Check for missing data in the dataframe for quantitative
Missing_data_count = df.isna().sum()
print("No. of missing data points in different columns:\n",Missing_data_count)

# In case of missing data impute the data based on the most common value in the
# column
for i in df.columns:
    column_index = df.columns.get_loc(i)
    mode = df.mode().iloc[:,column_index]
    print(mode[0])
    df[i] = df[i].fillna(mode[0])

# Assess the summary of aggregate measures for the data
df.describe()

# We need to see if there any outliers in the dataset for continuous variables
# we check that based on the quartiles.

continous_features = ['age','trestbps','chol','thalach','oldpeak']

#Calculate and identify outliers
def detect_outliers(df_out):
    """Returned the number of outliers for continuous variable based on 
    interquartile range.
    """
    for each_feature in df_out.columns:
        feature_data = df_out[each_feature]
        q_1 = np.percentile(feature_data, 25.)
        q_3 = np.percentile(feature_data, 75.)
        iqr = q_3 - q_1
        outlier_step = iqr
        outliers = feature_data[~((feature_data >= q_1 - outlier_step) \
                & (feature_data <= q_3 + outlier_step))].index.tolist()
        print(f"For the feature {each_feature}, the number of outliers is {len(outliers)}.")
detect_outliers(df[continous_features])
###-------------------------------------------------------------------------###
###                     Descriptive Data Analyis                            ###
###-------------------------------------------------------------------------###
#Set the parameters that control the general style of the plots.
sns.set_style('white')
sns.set_palette('twilight')
p1 = sns.countplot(data= df, x='sex',hue='num')
plt.legend(title='"Diagnosis', loc='upper left', labels=['No Heart Disease', 'Heart Disease'])
plt.ylabel('Number of Cases',fontsize=13)
plt.xlabel('Gender',fontsize=13)
p1.set_xticklabels(('Female', 'Male'))
sns.despine()
###-------------------------------------------------------------------------###

p2 = sns.countplot(data= df, x='cp',hue='num')
plt.legend(title='"Diagnosis', loc='upper left', labels=['No Heart Disease','Heart Disease'])
plt.ylabel('Number of Cases',fontsize=13)
plt.xlabel('Chest Pain',fontsize=13)
p2.set_xticklabels(('Typical Angina', 'Atypical Angina','Non - Angina Pain','Asymptomatic'))
sns.despine()
###-------------------------------------------------------------------------###

p3 = sns.countplot(data= df, x='fbs',hue='num',order=[1,0])
plt.legend(title='"Diagnosis', loc='upper left', labels=['No Heart Disease','Heart Disease'])
plt.ylabel('Number of Cases',fontsize=13)
plt.xlabel('Fasting Blood Sugar (>120 mg/dl)',fontsize=13)
p3.set_xticklabels(('True', 'False'))
sns.despine()
###-------------------------------------------------------------------------###

p4 = sns.countplot(data= df, x='restecg',hue='num',order=[1,0,2])
plt.legend(title='"Diagnosis', loc='upper left', labels=['No Heart Disease','Heart Disease'])
plt.ylabel('Number of Cases',fontsize=13)
plt.xlabel('Resting Electrocardiographic Results',fontsize=13)
p4.set_xticklabels(('Normal', 'Abnormal','Ventricular Hypertrophy'))
sns.despine()
###-------------------------------------------------------------------------###

p5 = sns.countplot(data= df, x='exang',hue='num',order=[1,0])
plt.legend(title='"Diagnosis', loc='upper left', labels=['No Heart Disease','Heart Disease'])
plt.ylabel('Number of Cases',fontsize=13)
plt.xlabel('Exercise Induced Angina',fontsize=13)
p5.set_xticklabels(('Yes', 'No'))
sns.despine()
###-------------------------------------------------------------------------###

p6 = sns.countplot(data= df, x='slope',hue='num',order=[3,2,1])
plt.legend(title='"Diagnosis', loc='upper left', labels=['No Heart Disease','Heart Disease'])
plt.ylabel('Number of Cases',fontsize=13)
plt.xlabel('The Slope Of The Peak Exercise',fontsize=13)
p6.set_xticklabels(('Downsloping', 'Flat','Upsloaping'))
sns.despine()
###-------------------------------------------------------------------------###

p7 = sns.countplot(data= df, x='thal',hue='num',order=[6,3,7])
plt.legend(title='"Diagnosis', loc='upper left', labels=['No Heart Disease','Heart Disease'])
plt.ylabel('Number of Cases',fontsize=13)
plt.xlabel('Type of Thalassemia',fontsize=13)
p7.set_xticklabels(('Downsloping', 'Flat','Upsloaping'))
sns.despine()
###-------------------------------------------------------------------------###

p8 = sns.countplot(data= df, x='ca',hue='num',order=[3,2,1,0])
plt.legend(title='"Diagnosis', loc='upper left', labels=['No Heart Disease','Heart Disease'])
plt.ylabel('Number of Cases',fontsize=13)
plt.xlabel('The Number Of Major Blood Vessels',fontsize=13)
p8.set_xticklabels(('3', '2','1','0'))
sns.despine()
###-------------------------------------------------------------------------###

sns.histplot(data=df, x="age", kde=True, bins=20)
plt.title('Distribution of Age')
plt.xlabel('Age')
plt.ylabel('Number of Cases')
sns.despine()
###-------------------------------------------------------------------------###

sns.histplot(data=df, x="chol", kde=True, bins=20)
plt.title('Distribution of Cholesterol')
plt.xlabel('Cholesterol')
plt.ylabel('Number of Cases')
sns.despine()
###-------------------------------------------------------------------------###

sns.histplot(data=df, x="trestbps", kde=True, bins=20)
plt.title('Distribution of Resting Blood Pressure')
plt.xlabel('Resting Blood Pressure')
plt.ylabel('Number of Cases')
sns.despine()
###-------------------------------------------------------------------------###

sns.histplot(data=df, x="thalach", kde=True, bins=20)
plt.title('Distribution of the Maximum Heart Rate')
plt.xlabel('Maximum Heart Rate')
plt.ylabel('Number of Cases')
sns.despine()
###-------------------------------------------------------------------------###

sns.histplot(data=df, x="oldpeak", kde=True, bins=20)
plt.title('Distribution of the ST Depression Induced by Exercise')
plt.xlabel('ST Depression Induced by Exercise')
plt.ylabel('Number of Cases')
sns.despine()
###-------------------------------------------------------------------------###

p=sns.boxplot(data=df,x='num',y='age')
plt.xlabel('Diagnosis')
plt.ylabel('Age')
plt.xlabel('Diag')
p.set_xticklabels(['No Heart Disease', 'Heart Disease'])
###-------------------------------------------------------------------------###

p=sns.boxplot(data=df,x='num',y='trestbps')
plt.xlabel('Diagnosis')
plt.ylabel('Resting Blood Pressure')
p.set_xticklabels(['No Heart Disease', 'Heart Disease'])
###-------------------------------------------------------------------------###

p=sns.boxplot(data=df,x='num',y='chol')
plt.xlabel('Diagnosis')
plt.ylabel('Cholesterol')
p.set_xticklabels(['No Heart Disease', 'Heart Disease'])
###-------------------------------------------------------------------------###

p=sns.boxplot(data=df,x='num',y='thalach')
plt.xlabel('Diagnosis')
plt.ylabel('Maximum Heart Rate Achieved')
p.set_xticklabels(['No Heart Disease', 'Heart Disease'])
###-------------------------------------------------------------------------###

p=sns.boxplot(data=df,x='num',y='oldpeak')
plt.xlabel('Diagnosis')
plt.ylabel('ST Depression Induced By\n Exercise Relative To Rest')
p.set_xticklabels(['No Heart Disease', 'Heart Disease'])
###-------------------------------------------------------------------------###
# Assess the correlation between variables in the dataset. Correlation between
# variables might affect the regression analysis. So, this information will be
# helpful to explain any unexpected results or for model selection
corrmat = df.corr()
matrix = np.triu(corrmat)
fig, ax = plt.subplots(figsize=(12, 12))
sns.heatmap(corrmat, square=True, annot=True, cmap="Purples", mask = matrix)
plt.show()
#save the plot
fig.savefig('./Correlation Heatmap', bbox_inches='tight')
###-------------------------------------------------------------------------###



#Pylint
# (base) C:\Users\pandapar\Desktop\Uni_courses_2022\2023\Python\project>pylint python_project_ap_sg_draft_1.py
#
#-------------------------------------------------------------------
#Your code has been rated at 10.00/10 (previous run: 9.92/10, +0.08)
